kubectl-plugin/pkg/cmd/job/job_submit.go

```diff
@@ -557,6 +566,22 @@ func (options *SubmitJobOptions) raySubmitCmd() ([]string, error) {
 		raySubmitCmd = append(raySubmitCmd, "--verbose")
 	}
 
+	if len(options.headNodeSelectors) > 0 {
```

Comment:
Hmm, does `ray job submit` support these two options?

Reply:
cc @MortalHappiness Would you mind shepherding this PR and making sure it's well tested? Thanks!

Reply:
I think it is already well tested because this PR reuses the `generation.WorkerGroup` object, which already has tests for `NodeSelectors`.

Reply:
@MortalHappiness L569 - L583 adds `--head-node-selectors` and `--worker-node-selectors` to `ray job submit` command if I understand correctly. Does `ray job submit` support these two options?

Reply:
No, `ray job submit` does not have those two flags. But I think it's worth adding them, because it's common for data scientists to select GPU nodes. We don't need to make all the flags identical to `ray job submit`.

Reply:
Hmm, if it doesn't exist, the `ray job submit` command will fail, right?